app/views/dashboard/index.html.slim

```diff
@@ -1,4 +1,53 @@
-h1 Dashboard
+h1 Simple Monitoring
 .row
   .col-xs-12.discovery-control
-    a.btn.btn-primary[href="/nodes"] Choose master
+    span #{Minion.count} Nodes found
+.nodes-container
+  table.table
+    thead
+      tr
+        th
+          | Id
+        th.text-center
+          | Status
+        th
+          | Hostname
+        th
+          | Role
+        th
+          | Highstate
+        th.text-center
+          | Master
+    tbody
+      - Minion.all.each do |m|
+        tr
+          th
+            | #{m.id}
+          td.text-center
+            - if m.role
+              i.fa.fa-check-circle-o.text-success.fa-2x aria-hidden="true"
+            - else
+              span.fa-stack aria-hidden="true"
+                i.fa.fa-circle.fa-stack-2x.text-success aria-hidden="true"
+                i.fa.fa-refresh.fa-stack-1x.fa-spin.fa-inverse aria-hidden="true"
+          td
+            | #{m.hostname}
+          td
+            | #{m.role}
+          td
+            | #{m.highstate}
+          td.text-center
+            - if m.role == "master"
+              = radio_button_tag "roles[master]", m.id, true, disabled: true
+            - elsif Minion.all.collect(&:role).include?("master")
```

Comment:
This creates N+1 queries but I have fixed that already [here](https://github.com/kubic-project/velum/blob/fix_minion_polling/app/views/nodes/index.html.slim#L14) so you can leave it as is. I will take care of it.